avocado/plugins/assets.py

```diff
@@ -0,0 +1,179 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2019
+# Authors: Willian Rampazzo <willianr@redhat.com>
+
+import ast
+import errno
+import json
+import os
+import tempfile
+
+from avocado.utils import process
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLICmd
+
+
+TEST_TEMPLATE = r"""
+from avocado import Test
+
+class FetchAssets(Test):
+    def test_fetch_assets(self):
+{fetch_assets}
+"""
+
+
+class FetchAssetHandler(ast.NodeVisitor):
+
+    def __init__(self, file_name):
+        self.file_name = file_name
+        # Right now, the class handles just 'fetch_assets' call.
+        # In the future, it can be extended to search for other call patterns.
+        self.pattern = 'fetch_asset'
+        self.calls = []
+        self.result = None
+
+        # create Abstract Syntax Tree from test source file
+        with open(self.file_name) as source_file:
+            self.tree = ast.parse(source_file.read(), self.file_name)
+
+        # build calls list
+        self.visit(self.tree)
+    # __init__()
+
+    def _build_call(self, node):
+        """
+        Build the call string from ast nodes.
+        :param node: Call root node
+        :type node: ast.Call
+        :return: Call prepared to be executed into a test class
+        :rtype: str
+        """
+        # we handle code indentation
+        call = "        self." + node.func.attr + "("
+        for a in node.args:
+            if isinstance(a, ast.Name):
+                return None
+            elif isinstance(a, ast.Str):
```

Comment:
Well on one side you return None when variable is used, while on the other hand you silently skip everything that is not string (including lists).

I'd suggest skipping all values that can't be parsed (`ast.Name`) while including all values that can be extracted (including lists), eventually checking we have at least the bare-minimum to execute the fetch_asset call (one argument or name kwarg; although name might not be enough without locations... still we can live with partial knowledge)...

Alternatively I'd suggest chickening-out any time we can't parse something, therefore `else: return None` (safest but we'd not download calls with `hashes` or other "unimportant" things specified via variable...)